pilot/pkg/xds/delta.go

```diff
@@ -318,12 +320,35 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
 	if con.proxy.SidecarScope != nil && con.proxy.SidecarScope.Version != request.Push.PushVersion {
 		s.computeProxyState(con.proxy, request)
 	}
-	return s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+
+	err := s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case dwr := <-dependentResourceChan:
+		request := &model.PushRequest{
+			Full:   true,
+			Push:   con.proxy.LastPushContext,
+			Reason: model.NewReasonStats(model.DependentRequest),
+			Start:  con.proxy.LastPushTime,
+			Delta: model.ResourceDelta{
```

Comment:
I think this can be empty?

Reply:
I wondered if empty was permissible; glad to know it is!